### DISCUSSION TITLE
Twig Error After Capistrano Deploy

### DISCUSSION THREAD

**Discussion by shandybp** [flyntwp/flynt#577] [Q&A]
I deployed my Bedrock project to a staging server using Capistrano, and everything was fine except for Flynt. I'm getting this error:

Uncaught TypeError: Flynt\Utils\TwigExtensionPlaceholderImage::renderPlaceholderImage(): Argument #1 ($width) must be of type float, null given

Source of error: /flynt/Components/BlockImageText/index.twig on line 19

I'm using the same code as the repo.
- PHP@8.1

Can someone please help? This is not broken on local.

**Comment by timohubois** [comment]
@shandybp the problem is that the width of the image is not null. Is the image which causes the problem maybe a .svg image which does not have a viewBox nor width and height attributes?

A quickfix can be do add a default value: 
Instead of `{{ placeholderImage(image.width, image.height, 'rgba(125, 125, 125, 0.1)') }}` 
use `{{ placeholderImage(image.width|default(1), image.height|default(1), 'rgba(125, 125, 125, 0.1)') }}`. 

This is not perfect and can cause to wrong ratios and I would prefer to identify the source of problem.

**Reply by shandybp** [reply]
@timohubois 
I believe this error is hitting any component that uses the placeholder image. If the call to the image is in a  {% if image.post_mime_type == 'image/svg+xml' %} conditional, wouldn't that `suffice?` My BlockImageText component is a duplicate of the repo's component. I tried deleting the component, and next deploy the error was for a different component that used the TwigExtensionPlaceholder code.

But the test page I have uses an svg image uploaded to the media library and the image's svg code is below:

`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="240" height="240" fill="currentColor"><path d="M9 17C9 17 16 18 19 21H20C20.5523 21 21 20.5523 21 20V13.937C21.8626 13.715 22.5 12.9319 22.5 12C22.5 11.0681 21.8626 10.285 21 10.063V4C21 3.44772 20.5523 3 20 3H19C16 6 9 7 9 7H5C3.89543 7 3 7.89543 3 9V15C3 16.1046 3.89543 17 5 17H6L7 22H9V17ZM11 8.6612C11.6833 8.5146 12.5275 8.31193 13.4393 8.04373C15.1175 7.55014 17.25 6.77262 19 5.57458V18.4254C17.25 17.2274 15.1175 16.4499 13.4393 15.9563C12.5275 15.6881 11.6833 15.4854 11 15.3388V8.6612ZM5 9H9V15H5V9Z"/></svg> `

**Reply by timohubois** [reply]
@shandybp this is strange. We also use .svg images at flyntwp.com within a bedrock setup. Images from the media library should get transformed to timber objects where timber then[ tries to get the dimensions of the svg](https://github.com/timber/timber/blob/d56177f5c6a133d2b02b14b212e10c6176087ea7/src/ImageDimensions.php#L165). For some reasons it seems that this is currently not working. How do you render and pass the data to the component where the issue appears?

**Comment by shandybp** [comment]
@timohubois I am using the exact same "BlockImageText" component from your repo. I am wondering if my Capistrano deploy script is causing this, as I currently have to re-run composer from Flynt manually on the server, post-deploy- in order to generate the vendor directory.  Can you take a look? The theme tasks start at `# START THEME ------`
[deploy.txt](https://github.com/user-attachments/files/17662939/deploy.txt)
